crates/ruff_python_semantic/src/model.rs

```diff
@@ -1742,19 +1723,6 @@ impl<'a> SemanticModel<'a> {
             .intersects(SemanticModelFlags::NAMED_EXPRESSION_ASSIGNMENT)
     }
 
-    /// Return `true` if the model is in a comprehension assignment (e.g., `_ for x in y`).
-    pub const fn in_comprehension_assignment(&self) -> bool {
-        self.flags
-            .intersects(SemanticModelFlags::COMPREHENSION_ASSIGNMENT)
-    }
-
-    /// Return `true` if the model is visiting the r.h.s. of an `__all__` definition
-    /// (e.g. `"foo"` in `__all__ = ["foo"]`)
-    pub const fn in_dunder_all_definition(&self) -> bool {
-        self.flags
-            .intersects(SemanticModelFlags::DUNDER_ALL_DEFINITION)
-    }
```

Comment:
I think we should leave this as-is for any flags that still exist. It's nice to have a consistent interface even if unused. (But of course we can remove for any flags that we've removed.)